cmd/auth/login.go

```diff
@@ -3,26 +3,279 @@
 package auth
 
 import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
 	"github.com/pingidentity/pingcli/cmd/common"
+	auth_internal "github.com/pingidentity/pingcli/internal/auth"
+	auth_commands "github.com/pingidentity/pingcli/internal/commands/auth"
+	"github.com/pingidentity/pingcli/internal/configuration/options"
+	"github.com/pingidentity/pingcli/internal/customtypes"
+	"github.com/pingidentity/pingcli/internal/errs"
+	"github.com/pingidentity/pingcli/internal/output"
+	"github.com/pingidentity/pingcli/internal/profiles"
+	svcOAuth2 "github.com/pingidentity/pingone-go-client/oauth2"
 	"github.com/spf13/cobra"
+	"golang.org/x/oauth2"
+)
+
+var (
+	// ErrUnknownAuthMethod is returned when an unknown authentication method is specified
+	ErrUnknownAuthMethod = fmt.Errorf("unknown authentication method")
 )
 
+// NewLoginCommand creates a new login command that authenticates users using one of the supported
+// authentication flows: device code, authorization code, or client credentials
 func NewLoginCommand() *cobra.Command {
 	cmd := &cobra.Command{
 		Args:                  common.ExactArgs(0),
 		DisableFlagsInUseLine: true, // We write our own flags in @Use attribute
-		Long:                  "Login user to the CLI",
+		Long:                  "Login user to the CLI using one of the supported authentication flows: device code, authorization code, or client credentials",
 		RunE:                  authLoginRunE,
 		Short:                 "Login user to the CLI",
 		Use:                   "login [flags]",
 	}
 
+	cmd.Flags().AddFlag(options.AuthMethodAuthCodeOption.Flag)
+	cmd.Flags().AddFlag(options.AuthMethodClientCredentialsOption.Flag)
+	cmd.Flags().AddFlag(options.AuthMethodDeviceCodeOption.Flag)
+	cmd.Flags().AddFlag(options.AuthServiceOption.Flag)
+	cmd.Flags().AddFlag(options.AuthUseKeychainOption.Flag)
+
+	// Enforce that exactly one authentication method must be specified
+	cmd.MarkFlagsMutuallyExclusive(
+		options.AuthMethodAuthCodeOption.Flag.Name,
+		options.AuthMethodClientCredentialsOption.Flag.Name,
+		options.AuthMethodDeviceCodeOption.Flag.Name,
+	)
+
 	return cmd
 }
 
+// authLoginRunE implements the login command logic, handling authentication based on the selected
+// method (auth code, device code, or client credentials) with support for interactive configuration
 func authLoginRunE(cmd *cobra.Command, args []string) error {
-	// l := logger.Get()
-	// l.Debug().Msgf("Auth Login Subcommand Called.")
+	ctx := context.Background()
+
+	// Get current profile name for messaging
+	profileName, err := profiles.GetOptionValue(options.RootActiveProfileOption)
+	if err != nil {
+		return fmt.Errorf("failed to get active profile: %w", err)
```

Comment:
Commands should return PingCLIErrors, similar to other commands, not `fmt.Errorf`

Reply:
this is now updated